internal/services/databricks/databricks_virtual_network_peering_resource_test.go

```diff
@@ -0,0 +1,266 @@
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDatabricksVirtualNetworkPeering_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDatabricksVirtualNetworkPeering_completeUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_databricks_virtual_network_peering", "test")
+	r := DatabricksVirtualNetworkPeeringResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.completeUpdate(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (DatabricksVirtualNetworkPeeringResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.DataBricks.VnetPeeringClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("making Read request on Databricks %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (DatabricksVirtualNetworkPeeringResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-databricks-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_virtual_network" "remote" {
+  name                = "acctest-vnet-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  address_space       = ["10.0.1.0/24"]
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_databricks_workspace" "test" {
+  name                = "acctest-ws-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  sku                 = "standard"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (DatabricksVirtualNetworkPeeringResource) update(data acceptance.TestData) string {
+	template := DatabricksVirtualNetworkPeeringResource{}.template(data)
```

Comment:
we can give this method a name of the type and call the other method directly fwiw:

```suggestion
func (r DatabricksVirtualNetworkPeeringResource) update(data acceptance.TestData) string {
	template := r.template(data)
```

Reply:
Fixed.